File: Plugins/Lpp.Dns.General.Metadata/Content/Models/Terms/RxAmt.ts
/// <reference path="../../../../../Lpp.Mvc.Composition/Lpp.Mvc.Boilerplate/jsBootstrap.d.ts" />
/// <reference path="../Terms.ts" />

module DataCheckerModels {
    export interface IRxAmtTermData extends RequestCriteriaModels.ITermData {
        RxAmounts: RxAmtTypes[];
    }

    export enum RxAmtTypes {
        LessThanZero = 0,
        Zero = 1,
        TwoThroughThirty = 2,
        Thirty = 3,
        Sixty = 4,
        Ninety = 5,
        OneHundredTwenty = 6,
        OneHundredEighty = 7,
        Other = 8,
        Missing = 9
    }
} 

File: PopMedNet.vNext/Plugins/Lpp.Dns.General.CriteriaGroupCommon/Models/Terms/RxAmt.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.Text;

namespace RequestCriteria.Models
{
    [DataContract]
    public class RxAmountData : TermData
    {
        [DataMember]
        public RxAmountTypes[] RxAmounts { get; set; }
    }

    [DataContract]
    public enum RxAmountTypes
    {
        [EnumMember]
        LessThanZero = 0,      // <0
        [EnumMember]
        Zero = 1,              // 0-1
        [EnumMember]
        TwoThroughThirty = 2,  // 2-30
        [EnumMember]
        Thirty = 3,            // 31-60
        [EnumMember]
        Sixty = 4,             // 61-90
        [EnumMember]
        Ninety = 5,            // 91-120
        [EnumMember]
        OneHundredTwenty = 6,  // 121-180
        [EnumMember]
        OneHundredEighty = 7,  // >180
        [EnumMember]
        Other = 8,
        [EnumMember]
        Missing = 9
    }
}
